Extending/Packages/Good-practice/index.md

```diff
@@ -0,0 +1,161 @@
+---
+versionFrom: 9.0.0
+meta.Title: "Package development good practice"
+meta.Description: "Information on common defaults and good practice for umbraco package development"
+---
+
+# Good practice and defaults
+_This document provides some guides and notes on package development, and good practice that will help you maintain and support your package through multiple releases and versions of Umbraco. These good practices are not prescriptive, but mearly offer a guide as to what we have found works well, and not-so-well, when developing packages for Umbraco_
+
+## App_Plugins
+Files in the `App_Plugins` folder should be considered immutable, that is they are not something a user of your package should be expecting to change on their site. 
+
+The default delivery method for files to the App_Plugins folder is via a `.targets` file within a package, this means when a website is built the files in this folder are copied over from the nuget cache, and any changes a user might have made will be lost. Equally if the user performs a `dotnet clean` on a solution all files in the App_Plugins folder will likely be deleted. 
+
+If you have files that you expect users of your package to alter you should not place them in the `App_Plugins` folder.
+
+## Views
+Views are used to render content on the front end of a website. If your plugin provides a way for the user to present the content publicly you should copy these files to the views folder. 
+
+As the files will still be copied during build you should ensure your target file doesn't overwrite newer or altered files and doesn't delete files on clean. 
+
+_This method isn't perfect and if you do have a package that is heavy on views you might consider building a dotnet template or Razor class library to contain the views so they don't pollute your user's solutions._ 
```

Comment:
Another way is to use ASP.NET [Tag Helpers](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/intro) or [View Components](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-components) instead. Or maybe even better: include the views in a Razor Class Library instead?

Reply:
yes this is a better way (RCL) but until v10 this isn't smooth enough - i suspect events will overtake this document 😄 and we should change it to how to do this in a RCL.

Reply:
RCL projects have only been properly supported since v11, so it makes sense to only include additional information about how and when to use that project type in a version 11 specific documentation page...